packages/openapi-v2/src/controller-spec.ts

```diff
@@ -177,17 +206,84 @@ function resolveControllerSpec(constructor: Function): ControllerSpec {
  */
 export function getControllerSpec(constructor: Function): ControllerSpec {
   let spec = MetadataInspector.getClassMetadata<ControllerSpec>(
-    REST_API_SPEC_KEY,
+    REST_CONTROLLER_SPEC_KEY,
     constructor,
     {ownMetadataOnly: true},
   );
   if (!spec) {
     spec = resolveControllerSpec(constructor);
-    MetadataInspector.defineMetadata(REST_API_SPEC_KEY, spec, constructor);
+    MetadataInspector.defineMetadata(
+      REST_CONTROLLER_SPEC_KEY,
+      spec,
+      constructor,
+    );
   }
   return spec;
 }
 
+export function jsonToSchemaObject(json: JsonDefinition): SchemaObject {
+  const emptySchemaObj: SchemaObject = {};
+  // tslint:disable-next-line:no-any
+  const def: {[key: string]: any} = json;
+  for (const property in def) {
+    const val = def[property];
+    switch (property) {
+      // converts excepted properties to SchemaObject definitions
+      case 'type': {
+        if (def.type === 'array' && !def.items) {
+          throw new Error(
+            '"items" property must be present if "type" is an array',
+          );
+        }
+        emptySchemaObj.type = Array.isArray(json.type)
+          ? json.type[0]
+          : json.type;
+        break;
+      }
+      case 'allOf': {
+        const collector: SchemaObject[] = [];
+        for (const item of def.allOf) {
```

Comment:
```ts
emptySchemaObj.allOf = def.allOf.map(item => jsonToSchemaObject(item));
```

I think `emptySchemaObj` is not a very good name because `emptySchemaObj` is no longer empty after this line. Perhaps `target` or `result` would be better one?